src/block_cache/block_cache.c

```diff
@@ -415,43 +396,25 @@ __wt_blkcache_put(WT_SESSION_IMPL *session, WT_ITEM *data, const uint8_t *addr,
 
     blkcache = &S2C(session)->blkcache;
     blkcache_store = NULL;
-    data_ptr = NULL;
-
-    /* Bypass on write if the no-write-allocate setting is on */
-    if (write && blkcache->cache_on_writes == false) {
-        WT_STAT_CONN_INCR(session, block_cache_bypass_writealloc);
-        return (0);
-    }
```

Comment:
Why did you choose to remove this?

Reply:
I moved it into the caller, since it's a cheap test to make to avoid the function call. I did the same thing with some of the other checks for "don't call __wt_blkcache_put()".